src/Utilities/functions.jl

```diff
@@ -796,6 +810,10 @@ is_canonical(::MOI.AbstractFunction) = false
 
 is_canonical(::Union{MOI.VariableIndex,MOI.VectorOfVariables}) = true
 
+function is_canonical(f::MOI.ScalarNonlinearFunction)
+    return all(is_canonical(arg) for arg in f.args)
```

Comment:
It seems we now need to declare `is_canonical(::Number) = true` ^^

Reply:
Haha. I never run the full tests locally, just the file I'm working on.

I also switched to a non-recursive implementation, since this will get called for all `add_constraint`.